src/main.rs

```diff
@@ -16,52 +12,57 @@ struct Config {
     mail_from: String,
     mails_to: Vec<String>,
     subject: String,
-    body: String
+    body: String,
 }
 #[derive(Serialize)]
 struct Context {
     first_name: String,
-    last_name: String
+    last_name: String,
 }
 
-
 fn main() {
-    let config_file = std::fs::File::open("config.yaml").unwrap();
-    let config_file_btree: Config = serde_yaml::from_reader(config_file).unwrap();
+    let config_file = File::open("config.yaml").expect("Can't open file");
+    let mut config: Config = serde_yaml::from_reader(config_file).expect("Can't parse opened file");
     let mut body_template = TinyTemplate::new();
 
+    config.mails_to.shuffle(&mut thread_rng());
 
-    let mut mailts_to: Vec<String> = config_file_btree.mails_to;
-    mailts_to.shuffle(&mut thread_rng());
-
-    body_template.add_template("body", &config_file_btree.body).unwrap();
+    body_template
+        .add_template("body", &config.body)
+        .expect("Can't add body to the template");
 
-    for (i, mail_to) in mailts_to.iter().enumerate() {
-        let mut receiving: usize = 0;
+    for (i, mail_to) in config.mails_to.iter().enumerate() {
+        let mut receiving_gift_index = i + 1;
 
-        // use to take the gift receive mail in the Vec if it is the last mails get first Vec entry
-        if i+1 < mailts_to.len() {
-            receiving = i+1;
+        // take the first mail_to if the current is the last
+        if i == config.mails_to.len() - 1 {
+            receiving_gift_index = 0;
         }
 
-        let chunks_receiving_mail: Vec<&str> = mailts_to[receiving].split(" ").collect();
+        let chunks_receiving_mail: Vec<&str> =
+            config.mails_to[receiving_gift_index].split(' ').collect();
         let context = Context {
             first_name: chunks_receiving_mail[0].to_string(),
-            last_name: chunks_receiving_mail[1].to_string()
+            last_name: chunks_receiving_mail[1].to_string(),
         };
 
         let email = Message::builder()
-            .from(config_file_btree.mail_from.parse().unwrap())
-            .to(mail_to.parse().unwrap())
-            .subject(config_file_btree.subject.to_string())
-            .body(body_template.render("body", &context).unwrap().to_string())
-            .unwrap();
+            .from(config.mail_from.parse().expect("Can't parse mail_from"))
+            .to(mail_to.parse().expect("Can't parse mail_to"))
+            .subject(config.subject.clone())
```

Comment:
Alors ici plutot que d'utiliser `to_string()` sur une string il est preferable d'utiliser `clone()`, le resultat est strictement identique mais tu es beaucoup plus explicite sur ce que tu souhaites faire. 

Maintenant pourquoi le resultat est identique ? Merci de me la poser, c'est une excellente question !
Car dans l'implementation de String on retrouve : 
```rust
impl ToString for String {
    #[inline]
    fn to_string(&self) -> String {
        self.to_owned()
    }
}
```
Et pour ToOwned for T (dans notre cas le type generique T est String)
```rust
impl<T> ToOwned for T
where
    T: Clone,
{
    type Owned = T;
    fn to_owned(&self) -> T {
        self.clone()
    }
    ...
}
```
Donc `to_string` sur une String revient a faire un `clone()`, mais c'est vachement plus lisible de faire directement un `clone()` car ce que tu veux au final c'est recuperer le ownership sur ta donnee, ce que permet le clone en te creant un nouvel espace memoire avec une nouvelle donnee indentique et rien que pour toi :)